### PR TITLE
Compute orientation based on velocity

### PR DESCRIPTION
**Pull request by daniel-holz**

### CODE REVIEW

**Review by LauWerUnity — APPROVED** [review]

File: Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs [hunk 1]
```diff
@@ -22,22 +21,54 @@ public void Execute(in ArchetypeChunk chunk, int unfilteredChunkIndex, bool useE
...
 
+            // derive heading of velocity
+            var heading = math.atan2(velNew.x, velNew.z);
+            var headingRot = Quaternion.AngleAxis(math.degrees(heading), Vector3.up);
+
+            // derive pitch of velocity, by bringing velocity into local space of object (with z forward) and
+            // calculating the angle of the velocity in the local yz plane of the object
+            var velForwardLocal = math.rotate(math.inverse(headingRot), velNew);
+            var pitch = math.atan2(velForwardLocal.y, velForwardLocal.z);
+            var pitchRot = Quaternion.AngleAxis(math.degrees(pitch), Vector3.left);
```
Comment (Impossible) [inline Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs:67]: Just FYI this is using UnityEngine Quaternion, and for consistent Unity.Mathematics usage you want to use lower case quaternion instead https://docs.unity3d.com/Packages/com.unity.mathematics@1.0/api/Unity.Mathematics.quaternion.html#Unity_Mathematics_quaternion_AxisAngle_Unity_Mathematics_float3_System_Single_

File: Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs [hunk 2]
```diff
@@ -22,22 +21,54 @@ public void Execute(in ArchetypeChunk chunk, int unfilteredChunkIndex, bool useE
...
+
+            // stop at lower bottom
+            pos.y *= notOnGround;
 
             tmComp.Value.Position = pos;
             velComp.Value = velNew;
 
+            // derive heading of velocity
+            var heading = math.atan2(velNew.x, velNew.z);
+            var headingRot = Quaternion.AngleAxis(math.degrees(heading), Vector3.up);
```
Comment (Impossible) [inline Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs:61]: For up/down/left/right you can use math.up, math.left, etc.  UnityEngine math types are still structs, so they work fine in Burst, but it is preferred to use Unity.Mathematics as much as possible  in Burst compiled methods (this is not always possible, especially with APIs in UnityEngine)

Comment (daniel-holz) [reply Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs:61]: Good to know. Thanks.

File: Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs (hunk 1 above)
Comment (daniel-holz) [reply Ported/combat-bees/Assets/Scripts/Systems/MovementSystem.cs:67]: Good one! I was looking for that yesterday and found that Quaternion function first 😅

**Pull request merged**